config/v1/types_kmsencryption.go

```diff
@@ -0,0 +1,28 @@
+package v1
+
+// KMSConfig defines the configuration for the KMS instance
+// that will be used with KMSv2 encryption
+//
+// +openshift:enable:FeatureGate=KMSv2
+type KMSConfig struct {
```

Comment:
What expansion do you expect in the future? Does this need to be a union with a platform type as the discriminator?

Reply:
Yes, so I ended up adding a provider type with similar unionDiscriminator + related CEL validation.

In theory, we could have reused PlatformType which is used in Infrastructure status but I do not want to couple it immediately with all cloud providers (as 1. we want to support AWS KMS only rn, 2. not all platform providers may have KMSes).

Reply:
I agree with not re-using the constant, and keeping the enum restricted, that's good. I wonder though if we should re-use the name here, so from a user perspective, it's still `platformType` that they are setting, WDYT?

Reply:
Yeah, but in the future we might have requests to add KMS provider(s) which are not coherent with our PlatformType offerings eg. Vault or an XYZ HSM, etc. that led me to not use the `platformType` name from an UX perspective.

Reply:
Then perhaps we should have the enum value determine the type of the KMS, and not just "AWS"? 

Would it be possible to use the AWS implementation on an Azure cluster?

Reply:
> Would it be possible to use the AWS implementation on an Azure cluster?

Generally, the KMS instance resides outside of the control-plane so that could be a use case (however, I don't think many users may ask for it - so we can decide ourselves later how we want to support that).

On the contrary, to answer that still think that the answer is a YES, if we flip the scenario say: we can have a new provider type in the future eg. Vault that is supported on both AWS, Azure or any cloud provider for that matter.

Reply:
> Then perhaps we should have the enum value determine the type of the KMS, and not just "AWS"?

Din't quite get you on this one. Here's what is currently being used as the enum:
```go
// KMSProviderType is a specific supported KMS provider
// +kubebuilder:validation:Enum="";AWS
type KMSProviderType string
```

Are you suggesting to do it differently?